contracts/zerodev/account/kernel/WalletKernel.sol

```diff
@@ -0,0 +1,189 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@gnosis.pm/safe-contracts/contracts/GnosisSafe.sol";
+import "@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol";
+import "../../plugin/IPlugin.sol";
+import "../../../core/Helpers.sol";
+import "../../../interfaces/IAccount.sol";
+import "../../utils/OpinionatedExec.sol";
+import "./Compatibility.sol";
+
+struct WalletKernelStorage {
+    address owner;
+    uint256 nonce;
+}
+
+
+/// @title Wallet Kernel
+/// @author taek<leekt216@gmail.com>
+/// @notice wallet kernel for minimal wallet functionality
+/// @dev supports only 1 owner and 1 threshold, multiple plugins
+contract WalletKernel is IAccount, EIP712, Compatibility {
+
+    function getWalletKernelStorage() internal pure returns (WalletKernelStorage storage ws) {
+        bytes32 storagePosition = bytes32(uint256(keccak256("zero-dev.account.walletkernel")) - 1);
+        assembly {
+            ws.slot := storagePosition
+        }
+    }
+
+    uint256 constant private SIG_VALIDATION_FAILED = 1;
+
+    address public immutable entryPoint;
+
+    error QueryResult(bytes result);
+
+    constructor(address _entryPoint) EIP712("WalletKernel", "0.0.1") {
```

Comment:
Not sure if I'm missing something obvious but how will the `owner` actually be set?

Reply:
ah we don't have that in this draft sorry it will be done with initialize() function though